### PR TITLE
Documentation update

### PR DESCRIPTION
**Pull request by alicerdv**
## What is the goal of this PR?

Update doc strings and update sphinx documentation to reflect the new changes we have recently introduced such as the addition of the Pipeline and Oracle

## What are the changes implemented in this PR?

Update the sphinx conf.py file so that
1) Doc strings from __init__ are rendered (`autoclass_content = "both"`)
2) The path of the class is not displayed, only the name of the class (`add_module_names = False`)

Update the intro and examples in the sphinx documentation to reflect the new changes we have recently made
Update the sphinx documentation so that the pipeline and oracle modules are rendered
Update missing doc strings

### CODE REVIEW

**Comment by alicerdv** [comment]
This isn't finished yet, it will need to be updated after any name change PRs are merged

**Comment by alicerdv** [comment]
Some of the images/code examples might need to be updated to align with the paper. I wasn't sure which images we wanted to keep

**Review by a-pouplin — DISMISSED** [review]
Nice ! Everything seems good on my side :)

File: docs/source/notes/quick_start.rst [hunk 1]
```diff
@@ -1,10 +1,9 @@
...
-As discussed in the :ref:`whatisal` section, the **PyRelationAL** package decomposes the active learning workflow into four
-main components: 1) a data manager, 2) a model, 3) an acquisition strategy built around an informativeness measure,
-and 4) an oracle. Note however, that the oracle is external to the package. In this section, we work through an example
-to illustrate how to instantiate and combine a data manager, a model, and an acquisition strategy.
+As discussed in the :ref:`whatisal` section, the **PyRelationAL** package decomposes the active learning workflow into five
+main components: 1) a data manager, 2) a model manager, 3) an acquisition strategy built around an informativeness measure, 4) an oracle and 5) a pipeline.
+In this section, we work through an example to illustrate how to instantiate and combine a data manager, a model manager, an acquisition strategy and an oracle.
 
 Data Manager
 ------------
```
Comment (thomasgaudelet) [inline docs/source/notes/quick_start.rst:9]: we should add a mention here that the pyrelational datamanager makes some assumptions about dataset. e.g

- it assumes that the labels are contained in an independent attribute of the dataset (unlike TensorDatasets) that can be specified with the `label_attr` input to the data manager
- that the dataset __getitem __ return a tuple of tensors
- a number of methods assumes the featurization is entirely contained in a single tensor, corresponding to the first element of the tuple returned by the __getitem __

Comment (thomasgaudelet) [reply docs/source/notes/quick_start.rst:9]: maybe to add to the readme too

File: docs/Makefile [hunk 2]
```diff
@@ -5,7 +5,7 @@
 # from the environment for the first two.
 SPHINXOPTS    ?=
 SPHINXBUILD   ?= sphinx-build
-SPHINXPROJ    = RelationAL
```
Comment (paulmorio) [inline docs/Makefile:8]: Oh how I wish we were able to get that :D. Thanks for spotting this.

File: pyrelational/model_managers/abstract_model_manager.py [hunk 3]
```diff
@@ -9,16 +9,18 @@
 
 
 class ModelManager(ABC, Generic[ModelType, E]):
+    """
+    Abstract class used to wrap models to interact with the ActiveLearningStrategy modules.
```
Comment (paulmorio) [inline pyrelational/model_managers/abstract_model_manager.py:13]: I think this needs to be updated to `ActiveLearningStrategy` -> `Strategy`

File: pyrelational/model_managers/ensemble_model_manager.py [hunk 4]
```diff
@@ -27,15 +27,24 @@ def __init__(
...
         self.device = _determine_device(self.trainer_config.get("gpus", 0))
         self.n_estimators = n_estimators
 
     def __call__(self, loader: DataLoader[Any]) -> torch.Tensor:
         """
+        Call method to output model predictions for each model in the ensemble
 
         :param loader: pytorch dataloader
-        :return: model predictions
+        :return: model predictions of shape (n_estimators, number of samples in loader, 1)
```
Comment (paulmorio) [inline pyrelational/model_managers/ensemble_model_manager.py:47]: Thanks for adding this detail!

File: docs/source/reference/strategies.rst [hunk 5]
```diff
@@ -4,28 +4,38 @@ pyrelational.strategies
...
 
 Strategies for regression tasks
 -------------------------------
 
+Abstract regression strategy
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. automodule:: pyrelational.strategies.regression.abstract_regression_strategy
+   :members:
+   :undoc-members:
+   :special-members: __call__
```
Comment (paulmorio) [inline docs/source/reference/strategies.rst:20]: This is good

File: pyrelational/strategies/regression/bald_strategy.py [hunk 6]
```diff
@@ -29,13 +29,30 @@ def __init__(
         self,
         temperature: float = 0.5,
     ):
+        """
+        :param temperature: parameter greater than 0 to divide scores before applying softmax.
+            A lower temperature parameter will give a peakier distribution of probabilities
```
Comment (paulmorio) [inline pyrelational/strategies/regression/bald_strategy.py:34]: I had to look up if peakier is a proper description. In some circles it is! I learned something today.

**Review by paulmorio — CHANGES_REQUESTED** [review]
Thanks for the hard work and all the clarifications on the method, its a huge improvement for the library. I just added one comment where I saw some reference to a deprecated module. Should approve after.

File: docs/source/notes/using_your_own_strategy.rst [hunk 7]
```diff
@@ -71,18 +68,18 @@ random from the remaining queryable set.
...
-            super(EpsilonGreedyStrategy, self).__init__(datamanager, model)
+        def __init(self):
+            super(EpsilonGreedyStrategy, self).__init__()
 
-        def suggest(self, num_annotate, eps=0.05):
+        def __call__(self, num_annotate, data_manager, model_manager, eps=0.05):
             assert 0 <= eps <= 1, "epsilon should be a float between 0 and 1"
-            self.model.train(self.l_loader, self.valid_loader)
-            output = self.model(self.u_loader)
+            model_manager.train(data_manager.get_labelled_loader(), data_manager.get_validation_loader())
```
Comment (thomasgaudelet) [inline docs/source/notes/using_your_own_strategy.rst:76]: train_and_infer might be useful here?

File: pyrelational/strategies/regression/bald_strategy.py (hunk 6 above)
Comment (thomasgaudelet) [reply pyrelational/strategies/regression/bald_strategy.py:34]: It is a weird word one for sure

File: pyrelational/model_managers/abstract_model_manager.py (hunk 3 above)
Comment (alicerdv) [reply pyrelational/model_managers/abstract_model_manager.py]: thanks, I have changed it now :)

File: docs/source/notes/using_your_own_strategy.rst (hunk 7 above)
Comment (alicerdv) [reply docs/source/notes/using_your_own_strategy.rst]: I've updated it now

File: docs/source/notes/quick_start.rst (hunk 1 above)
Comment (alicerdv) [reply docs/source/notes/quick_start.rst]: good points, I have added it here, in the doc string and in the readme

**Review by thomasgaudelet — APPROVED** [review]

**Review by paulmorio — APPROVED** [review]
Approved
